pym/bob/generators/QtCreatorGenerator.py

```diff
@@ -169,28 +171,36 @@ def generateQtProject(package, destination, updateOnly, projectName, includeDirs
         newPath = os.path.join(symlinkDir, name)
         if not updateOnly: os.symlink(os.path.join(os.getcwd(), path), newPath)
         for root, directories, filenames in os.walk(newPath):
+            hasInclude = False
+            if ('.git' in root) or ('.svn' in root):
+                continue
             for filename in filenames:
                 if source.match(filename) or cmake.match(filename) or filename == 'CMakeLists.txt':
                     sList.append(os.path.join(os.getcwd(), os.path.join(root,filename)))
                 if filter and additionalFiles.match(filename):
                     sList.append(os.path.join(os.getcwd(), os.path.join(root,filename)))
-            # it's faster to add all directories to includes and not to use regex, sort & remove duplicate entries
-            # this also helps qt-creator to resolve includes like <linux/bitops.h> which
-            # is not found in case there is no header in 'linux'
-            if not '.git' in root and not '.svn' in root:
-               hList.append(os.path.join(os.getcwd(),root))
-
-    for i in includeDirs:
-        if os.path.exists(i):
-            for root, directories, filenames in os.walk(i):
-                hList.append(os.path.join(i,root))
+                if not hasInclude and include.match(filename):
+                    hasInclude = True
+            if hasInclude:
+                oldPath = ""
+                # need to recursively add all directories from the include up to cwd to get includes like
+                # <a/b/c.h> resolved even if there is no include in 'a'
+                root = root[len(os.getcwd()):]
```

Comment:
Isn't `os.getcwd()` wrong here? The base directory is `newPath` which could be a totally different directory if the user specified "--destination"...